tests/v2/validation/provisioning/provisioning_rke2_ace_test.go

```diff
@@ -0,0 +1,237 @@
+			"worker":       false,
+		},
+		{
+			"controlplane": true,
+			"etcd":         false,
+			"worker":       false,
+		},
+		{
+			"controlplane": false,
+			"etcd":         false,
+			"worker":       true,
+		},
+		{
+			"controlplane": false,
+			"etcd":         false,
+			"worker":       true,
+		},
+	}
+
+	tests := []struct {
+		name      string
+		nodeRoles []map[string]bool
+		client    *rancher.Client
+	}{
+		{"Single Control Plane", nodeRoles0, r.client},
+		{"Multiple Control Planes", nodeRoles1, r.client},
+	}
+
+	var name string
+	for _, tt := range tests {
+		for _, kubeVersion := range r.kubernetesVersions {
+			name = tt.name + " Kubernetes version: " + kubeVersion
+			for _, cni := range r.cnis {
+				name += " cni: " + cni
+				r.Run(name, func() {
+					testSession := session.NewSession(r.T())
+					defer testSession.Cleanup()
+
+					testSessionClient, err := tt.client.WithSession(testSession)
+					require.NoError(r.T(), err)
+
+					clusterName := AppendRandomString(provider.Name)
+					generatedPoolName := fmt.Sprintf("nc-%s-pool1-", clusterName)
+					machinePoolConfig := provider.MachinePoolFunc(generatedPoolName, namespace)
+
+					machineConfigResp, err := machinepools.CreateMachineConfig(provider.MachineConfig, machinePoolConfig, testSessionClient)
+					require.NoError(r.T(), err)
+
+					machinePools := machinepools.RKEMachinePoolSetup(tt.nodeRoles, machineConfigResp)
+
+					cluster := clusters.NewRKE2ClusterConfig(clusterName, namespace, cni, cloudCredential.ID, kubeVersion, machinePools)
+
+					cluster.Spec.LocalClusterAuthEndpoint = r.localClusterAuthEndpoint
+
+					clusterResp, err := clusters.CreateRKE2Cluster(testSessionClient, cluster)
+					require.NoError(r.T(), err)
+
+					result, err := testSessionClient.Provisioning.Clusters(namespace).Watch(context.TODO(), metav1.ListOptions{
+						FieldSelector:  "metadata.name=" + clusterName,
+						TimeoutSeconds: &defaults.WatchTimeoutSeconds,
+					})
+					require.NoError(r.T(), err)
+
+					checkFunc := clusters.IsProvisioningClusterReady
+
+					err = wait.WatchWait(result, checkFunc)
+					assert.NoError(r.T(), err)
+					assert.Equal(r.T(), clusterName, clusterResp.Name)
+
+					clusterInfo, err := testSessionClient.Provisioning.Clusters(namespace).Get(context.TODO(), clusterResp.Name, metav1.GetOptions{})
+					require.NoError(r.T(), err)
+
+					clusterID := clusterInfo.Status.ClusterName
+
+					kubeConfig, err := kubeconfig.GetKubeconfig(r.client, clusterID)
+					require.NoError(r.T(), err)
+
+					original, err := r.client.SwitchContext(clusterResp.Name, kubeConfig)
+					require.NoError(r.T(), err)
+
+					originalResp, err := original.Resource(corev1.SchemeGroupVersion.WithResource("pods")).Namespace("").List(context.TODO(), metav1.ListOptions{})
+					require.NoError(r.T(), err)
+
+					for _, pod := range originalResp.Items {
+						r.T().Logf("Pod %v", pod.GetName())
+					}
+
+					contexts, err := kubeconfig.GetContexts(kubeConfig)
```

Comment:
If you don't need to the original context switch, then this call is only needed once. Does that make sense?